.github/workflows/build-test-javascript.yaml

```diff
@@ -97,7 +97,8 @@ jobs:
 
   upload-javascript-artifacts:
     needs: [build-test-js-artifacts]
-    if: github.ref == 'refs/heads/develop' || github.event.pull_request.head.repo.full_name == github.repository
+    # in english: (this pull request is not from a fork) AND (branch is "develop" OR branch starts with "release-" OR pull request has a "publish-js" label applied to it)
```

Comment:
Also what is this coding language? we should really go back to shell script and a Makefile.

Reply:
It's GitHub Actions' crazy expression language. It's not bad for simple stuff but it can get hairy quickly. See:
- https://docs.github.com/en/actions/learn-github-actions/expressions
- https://docs.github.com/en/actions/learn-github-actions/contexts
- https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idif